File: Project 01 - HelloWorld/HelloWorld-OC/HelloWorld/ViewController.m
//
//  ViewController.m
//  HelloWorld
//
//  Created by 孟昕欣 on 2019/8/10.
//  Copyright © 2019 Andy Meng. All rights reserved.
//

#import "ViewController.h"

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];

    // Update the view, if already loaded.
}

- (IBAction)didClickSayHelloButton:(id)sender {
    NSAlert *alert = [[NSAlert alloc]init]; // 定义alert变量
    [alert addButtonWithTitle:@"Ok"];       // 添加Ok按钮
    alert.messageText = @"ATourOfMacDev";    //弹窗内容
    alert.informativeText = @"Hello World !";   //描述性文字
    [alert beginSheetModalForWindow:self.view.window completionHandler:^(NSModalResponse returnCode){
        if(returnCode == NSAlertFirstButtonReturn){
            NSLog(@"this is OK Button tap");
        }
    }];
}

@end

File: Project 01 - HelloWorld/HelloWorld-Swift/HelloWorld/ViewController.swift
//
//  ViewController.swift
//  HelloWorld
//
//  Created by 孟昕欣 on 2019/8/10.
//  Copyright © 2019 Andy Meng. All rights reserved.
//

import Cocoa

class ViewController: NSViewController {

    @IBOutlet weak var sayHelloButton: NSButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }

    @IBAction func didClickSayHelloButton(_ sender: Any) {
        let alert = NSAlert();
        alert.alertStyle = NSAlert.Style.informational;
        alert.messageText = "ATourOfMacDev";
        alert.informativeText = "Hello World !";
        alert.addButton(withTitle: "OK");
//        alert.runModal();
        alert.beginSheetModal(for: view.window!, completionHandler: { res in
            print(res) // 按钮对应的 code 第一个是1000, 第二个 1001, 依次类推
        });
    }
    
}

